src/fhirtypes/ElementDefinition.ts

```diff
@@ -1864,7 +1885,18 @@ export class ElementDefinition {
           Type.Profile,
           Type.Extension
         );
-        if (json) {
+        const backboneProfile = this.hasProfileElementExtension();
+        if (backboneProfile) {
+          const def = StructureDefinition.fromJSON(json);
+          const targetElement = def.findElement(backboneProfile as string);
+          newElements = targetElement?.children().map(e => {
+            const eClone = e.clone();
+            eClone.id = eClone.id.replace(targetElement.id, this.id);
+            eClone.structDef = this.structDef;
+            eClone.captureOriginal();
+            return eClone;
+          });
+        } else if (json) {
```

Comment:
So as I understand it, this case is handling unfolding when the element isn't a content reference, so an example like:
```
Profile: MyPatient
Parent: Patient
* name.given 1..1

Profile: MyPatient2
Parent: Patient
* name ^type[0].profile[0] = http://example.org/StructureDefinition/MyPatient
* name ^type[0].profile[0].extension[0].url = http://hl7.org/fhir/StructureDefinition/elementdefinition-profile-element
* name ^type[0].profile[0].extension[0].valueString = "Patient.name"
* name.given 1..2
```
I thought it was still an open question on if we were going to allow this extension on non-`BackboneElement` type elements. Did we get clarity on if this extension can be used in that context? Very possible I am just forgetting the resolution of that.

Reply:
I think this is important to figure out, because if this extension can truly only be used on a `BackboneElement`, then the _only_ place we would have to unfold would be on a `contentReference`, so I think this code down here could actually go, and it would simplify things more. Now there is still an open question of if we would then want to add functionality so that we better handle applying constraints to elements that are already unfolded when type constraints are made, but I think that is a similarly open question for `only` rules, and so more of a separate task.

Reply:
> I thought it was still an open question on if we were going to allow this extension on non-`BackboneElement` type elements. Did we get clarity on if this extension can be used in that context? Very possible I am just forgetting the resolution of that.

[I posed the question on Zulip](https://chat.fhir.org/#narrow/stream/179177-conformance/topic/profile-element.20Extension/near/238329205) and we do want to scope the use of this extension to only include `BackboneElement`s. So, I've removed any unfolding that would've happened on non-contentReference elements. Because we haven't made a practice of altering the constraints on pre-existing children before, we've decided to leave the non-contentReference use of this extension for another day.